lib/ansible/utils/template.py

```diff
@@ -316,10 +324,18 @@ def my_finalize(thing):
 
 def template_from_string(basedir, data, vars, fail_on_undefined=False):
     ''' run a string through the (Jinja2) templating engine '''
-
     try:
         if type(data) == str:
             data = unicode(data, 'utf-8')
+   
+            # Check to see if the string we are trying to render is just referencing a single
+            # var.  In this case we don't wont to accidentally change the type of the variable
+            # to a string by using the jinja template renderer. We just want to pass it. 
+            only_one = SINGLE_VAR.match(data)
+            if only_one:
+                var_name = only_one.group(1)
+                if var_name in vars:
+                    return vars[var_name]
 
```

Comment:
Yea, I intentionally am only handling the use case of wanting to reference vars in a more complex structure and retaining type.  filters or other jinja operations would make it very hard to infer type correctly and seems more error prone without hooking deeper into Jinja's parsing logic.